lib/cloud_controller/dea/app_stager_task.rb

```diff
@@ -80,8 +103,8 @@ def handle_first_response(response, error, promise)
 
       def handle_second_response(response, error)
         @multi_message_bus_request.ignore_subsequent_responses
-        ensure_staging_is_current!
-        check_staging_error!(response, error)
+        ensure_staging_is_current!(false)
```

Comment:
It is similar but slightly different - there is no multi_message_bus_request in the http flow, and we pass in a boolean to ensure_staging_is_current to avoid checking the task_id (in the http flow we have to create a new app_stager_task, whereas the one for NATS is the same throughout the request lifecycle).  An alternative would have been to make check_task_id know whether we're using http or not and pick whether to use the current task's task_id or that from the message.  Would you prefer that over discarding a no-longer-current response in the controller upon receipt?